### PR TITLE
[58] Rewrite all IAM policies as documents

### LINKED ISSUE #58: Replace inline JSON policies with aws_iam_policy_documents for clarity
We have few places with a rather complex policy statement containing multiple levels of quoting, inlining and almost-templating. I think that tf code would look much clearer if we replace these policy stings with proper terraform data objects. The change will not affect any resource in AWS, it should be transparent to `plan`.

### PR DESCRIPTION
**Pull request by arturtamborski**
Closes #58 
This change does not require apply. It is technically a no-op from terraform state's point of view.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -12,6 +12,28 @@ provider "aws" {
   profile = "codeforpoznan"
 }
 
+data "aws_iam_policy_document" "codeforpoznan_public_policy_document" {
+  version = "2012-10-17"
+
+  statement {
+    sid    = "PublicListBucket"
+    effect = "Allow"
+    principals {
```
Comment (magul) [inline main.tf:21]: why these two statements differ in respect to `principals`?

File: main.tf [hunk 2]
```diff
@@ -12,6 +12,28 @@ provider "aws" {
   profile = "codeforpoznan"
 }
 
+data "aws_iam_policy_document" "codeforpoznan_public_policy_document" {
```
Comment (magul) [inline main.tf]: we should decide if we want to add `_document` suffix everywhere or omit it everywhere. Right now we have both approaches.

File: lambda/main.tf [hunk 3]
```diff
@@ -39,25 +39,22 @@ variable envvars {
...
-      },
-      "Effect": "Allow",
-      "Sid": ""
+data "aws_iam_policy_document" "role" {
+  version = "2012-10-17"
+
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    principals {
```
Comment (magul) [inline lambda/main.tf:49]: Did you test it? in JSON itself we have `Principal` (in singular) and here we have `principals` (in plural)

File: lambda/main.tf [hunk 4]
```diff
@@ -39,25 +39,22 @@ variable envvars {
...
-  "Statement": [
-    {
-      "Action": "sts:AssumeRole",
-      "Principal": {
-        "Service": [
-            "lambda.amazonaws.com"
-        ]
-      },
-      "Effect": "Allow",
-      "Sid": ""
```
Comment (magul) [inline lambda/main.tf:56]: `Sid` is unnecessary?

File: lambda/main.tf (hunk 3 above)
Comment (magul) [reply lambda/main.tf:49]: (the same comment applies to `Action` vs `actions`)

**Review by magul — COMMENTED** [review]
Some comments, but more or less looks fine!

File: lambda/main.tf (hunk 3 above)
Comment (arturtamborski) [reply lambda/main.tf:49]: `principals` and `actions` are terraform keywords, that's how it works. They will be converted to appropriate `Action` and `Principal` JSON keys.

File: lambda/main.tf (hunk 4 above)
Comment (arturtamborski) [reply lambda/main.tf:56]: the raw JSON had it empty, which is a default value for `sid` keyword in `document` block. I'll add `sid = ""` anyway to make it clear but yes, it's not needed.

Comment (magul) [reply lambda/main.tf:56]: I was just curious, you can leave it as it is.

File: main.tf (hunk 2 above)
Comment (arturtamborski) [reply main.tf]: will remove it

File: main.tf (hunk 1 above)
Comment (arturtamborski) [reply main.tf:21]: missed it, fixed

**Review by magul — APPROVED** [review]

**Pull request merged**